### PR TITLE
Adds a profile detail FE page

### PR DESCRIPTION
**Pull request by xyecs**
### What this PR does (required):
- Adds a profile detail FE page, also profiles interface along with other necessary changes

### Screenshots / Videos (front-end only):
https://user-images.githubusercontent.com/8740209/137642837-9082d243-dae0-4adb-971b-941b6582d964.mp4

### Any information needed to test this feature (required):
- N/A, just login -> profile (temporary)

### Any issues with the current functionality (optional):
- Request sent isn't linked to BE, completed in another issue.

### CODE REVIEW

File: client/src/pages/ProfileDetails/ProfileDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  const history = useHistory();
+  const initialDate = new Date();
+  const { firstName, lastName, location, hourlyRate, description, availability, photos } = mockProfile1;
+
+  const [startValue, setStartValue] = useState<Date | null>(new Date());
+  const [endValue, setEndValue] = useState<Date | null>(new Date());
+  const [sendRequest, setSendRequest] = useState<boolean>(false);
+  const [currPhotoPage, setCurrPhotoPage] = useState<number>(0);
+
+  console.log(photos);
```
Comment (ezeilo-su) [inline client/src/pages/ProfileDetails/ProfileDetails.tsx:26]: Let's remove this console log please

File: client/src/pages/ProfileDetails/ProfileDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  };
+
+  const handleLeftArrow = () => {
+    if (currPhotoPage > 0) {
+      setCurrPhotoPage(currPhotoPage - 1);
+    }
+  };
+
+  const handleSendRequest = () => {
+    // create new request here
```
Comment (ezeilo-su) [inline client/src/pages/ProfileDetails/ProfileDetails.tsx:48]: Let's remove this comment also

File: client/src/pages/ProfileDetails/ProfileDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    setSendRequest(true);
+  };
+
+  if (loggedInUser === undefined) return <CircularProgress />;
+  if (!loggedInUser) {
+    history.push('/login');
+    // loading for a split seconds until history.push works
```
Comment (ezeilo-su) [inline client/src/pages/ProfileDetails/ProfileDetails.tsx:55]: This comment too should be removed

**Review by moffatethan — APPROVED** [review]

File: client/src/pages/ProfileDetails/ProfileDetails.tsx (hunk 2 above)
Comment (xyecs) [reply client/src/pages/ProfileDetails/ProfileDetails.tsx]: resolved.

**Pull request merged**